Java/examples/src/main/java/com/amazon/randomcutforest/examples/parkservices/ForecastWithLimits.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.examples.parkservices;
+
+import static java.lang.Math.max;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import com.amazon.randomcutforest.config.Precision;
+import com.amazon.randomcutforest.config.TransformMethod;
+import com.amazon.randomcutforest.examples.Example;
+import com.amazon.randomcutforest.parkservices.ForecastDescriptor;
+import com.amazon.randomcutforest.parkservices.RCFCaster;
+import com.amazon.randomcutforest.parkservices.config.Calibration;
+import com.amazon.randomcutforest.returntypes.DiVector;
+import com.amazon.randomcutforest.returntypes.RangeVector;
+import com.amazon.randomcutforest.testutils.MultiDimDataWithKey;
+import com.amazon.randomcutforest.testutils.ShingledMultiDimDataWithKeys;
+
+/**
+ * The following example demonstrates the self calibration of RCFCast. Change
+ * various parameters -- we recommend keeping baseDimension = 1 (for single
+ * variate forecasting -- multivariate forecasting can be a complicated
```

Comment:
In RCF space, we are dealing with shingled input, right? baseDimension =1 and shingle size = 8 is the same thing for RCF as baseDimension = 8 and shingle size 1, right? Of course, with shingle size = 8, we are looking at past 8 in the history and our forecast would be more accurate. But other than that, there is no fundamental difference between baseDimension =1,shingle size = 8 and baseDimension = 8, shingle size =1, right?

Reply:
The issue is for forecast horizon = 5, the first would produce three vectors of size 5 (one value for each lookahead step, with upper and lower bounds). The second would be difficult to interpret because shingle size = 1 -- so there is no time-series aspect. For base dimension = 4 and shinglesize = 2, a forecast horizon of 5 would produce three vectors of length 4 *5 = 20.